### PR TITLE
Zz/simple update machine options

### PR DESCRIPTION
**Pull request by double-z**
- update machine_options when changed
- enable the ability to pass machine_options['transport_options']['options']
- allow for use of strings or symbols
- ensure host parameter is valid ip or resolvable hostname
- bump version to 0.0.8

### CODE REVIEW

**Comment by tyler-ball** [comment]
If you bump up the version, you need to make a point of releasing this change when you merge it to master.  Thats why I typically manage the release (changelog and version) separately from any specific PRs, because the changelog has to be updated for all commits since the last release.

File: lib/chef/provisioning/ssh_driver/driver.rb [hunk 1]
```diff
@@ -41,38 +41,50 @@ def self.canonicalize_url(driver_url, config)
...
+
+          if machine_spec.location && (machine_spec.location['driver_version'] != Chef::Provisioning::SshDriver::VERSION)
+            machine_spec.location['driver_version'] = Chef::Provisioning::SshDriver::VERSION
           end
+
         end
 
         def ready_machine(action_handler, machine_spec, machine_options)
-          ssh_machine = existing_ssh_machine(machine_spec)
+          allocate_machine(action_handler, machine_spec, machine_options)
```
Comment (tyler-ball) [inline lib/chef/provisioning/ssh_driver/driver.rb]: `ready_machine` isn't supposed to call `allocate_machine` - [chef-provisioning](https://github.com/chef/chef-provisioning/blob/master/lib/chef/provider/machine.rb#L36) takes care of this for you.

File: lib/chef/provisioning/ssh_driver/driver.rb [hunk 2]
```diff
@@ -41,38 +41,50 @@ def self.canonicalize_url(driver_url, config)
...
+          ssh_machine = existing_ssh_machine_to_sym(machine_spec)
 
           if !ssh_machine
             raise "SSH Machine #{machine_spec.name} does not have a machine file associated with it!"
           end
 
-          wait_for_transport(action_handler, ssh_machine, machine_spec, machine_options)
+          unless ssh_machine[:transport_options][:is_windows]
```
Comment (tyler-ball) [inline lib/chef/provisioning/ssh_driver/driver.rb]: Why aren't you waiting for transport if its a windows machine?

**Comment by double-z** [comment]
- additionally re-enabled the ability to use options for ssh
- clean up existing file on delete action
- add timestamp for modified and created times, the former of which is idempotent
- removed erroneous calls to allocate... they were left over from a registry thing i'm working on that needed it temporarily
- still need to add changelog and proper version flow, will have for full release, RC.1 is needed by Cisco

**Pull request merged**